src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,175 @@
+package duke;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+import duke.task.TaskList;
+
+import java.util.Scanner;
+
+/**
+ * Represents the user interface which handles input and output.
+ */
+public class Ui {
+    private Scanner input = new Scanner(System.in);
+
+    /**
+     * Read the input from the user.
+     *
+     * @return string representation of user's input in the CLI
+     */
+    public String readInput() {
+        return this.input.nextLine();
+    }
+
+
+    /**
+     * Close the input scanner.
+     */
+    public void closeInput() {
+        this.input.close();
+    }
+
+
+    /**
+     * Show greeting message to the user.
+     */
+    public void showGreet() {
+        String logo = " ____        _\n"
```

Comment:
Should `logo` be a constant in the class' attributes instead?

Reply:
Yep that's absolutely correct.. Should've made it static 🤦‍♂️